openwisp_network_topology/templates/netjsongraph/netjsongraph-script.html

```diff
@@ -1,7 +1,10 @@
 <script>
     loadNetJsonGraph = (el='body', url='{{ graph_url }}') => {
         const history_url = '{{ history_url }}';
-        const wsUrl = `ws://${window.location.host}${window.location.pathname}`;
+        const nonAdminView = {{ non_admin_view | yesno:"true,false" }};
+        const wsUrl = nonAdminView ?
+        `ws://${window.location.host}/admin${window.location.pathname}change/`:
+        `ws://${window.location.host}${window.location.pathname}`;
```

Comment:
Can you check whether the page is using HTTPS and in that case use `wss` please? This looks like a bug which we haven't caught yet because we aren't using this in prod yet (https://github.com/openwisp/ansible-openwisp2/issues/418).